### DISCUSSION TITLE
How to fail Nuclei command on CI/CD?

### DISCUSSION THREAD

**Discussion by Silva97** [projectdiscovery/nuclei#3218] [Q&A] [answered]
Hi, I want to run Nuclei on a CI/CD pipeline and fail the pipeline if the Nuclei match any vulnerability. Example:

```
nuclei -t my-templates -u example.com
```

The pipeline fail if the command-line returns exit status `!= 0`, but Nuclei always return exit status 0. It's has any option to make Nuclei returns exit status different of 0 if match any vulnerability?

**Comment by forgedhallpass** [comment, marked as answer]
As a workaround you can do something like the following:
```sh
nuclei -t my-templates -u example.com -silent -json > test.json && if [ -s test.json ]; then exit -1; fi
```
In this case nuclei would save the results in JSON format to the `test.json` file, the `if` condition would check if the file is empty and if not, it would exit with `-1`.

**Reply by Silva97** [reply]
Thank you!

**Comment by ehsandeep** [comment]
You may find this useful as well - https://escape.tech/blog/devsecops-part-iii-scanning-live-web-applications/
